File: Protogen/loxim_protocol/cpp/protocol/packages_data/Sint64Package.cpp
#include <string.h>
#include <stdlib.h>
#include "Sint64Package.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

Sint64Package::Sint64Package(\
		int64_t a_value
		)
{
	value=a_value;
}

Sint64Package::Sint64Package()
{
	value=0;
}

Sint64Package::~Sint64Package()
{
}

void Sint64Package::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	writer->writeInt64(value);
}

bool Sint64Package::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	value=reader->readInt64();
	return true;
}

bool Sint64Package::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	Sint64Package* p=(Sint64Package*)_package;
	if (!(getValue()==p->getValue())) return false;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages_data/Sint64Package.java
package pl.edu.mimuw.loxim.protocol.packages_data;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class Sint64Package extends Package {
		public final static long ID=(long)8;
	
    	/**
**/
    	private long value=0;
    	
	public Sint64Package()
	{
	}

		public Sint64Package(
		    	long a_value    	)
	{
        	value=a_value;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			value=reader.readInt64();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeInt64(value);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		Sint64Package p=(Sint64Package)obj;
	    if (!((value==p.getValue()))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public long getValue(){return value;};
		
    	public void setValue(long a_value){ value=a_value;};
		
    
	 
}


